controlplane/pkg/nsmd/nsmd.go

```diff
@@ -100,10 +101,27 @@ func (nsm *nsmServer) DeleteClientConnection(context context.Context, request *n
 	return &nsmdapi.DeleteConnectionReply{}, nil
 }
 
+func waitForDataplaneAvailable(model model.Model) {
+	logrus.Info("Waiting for dataplane available...")
+	ticker := time.NewTicker(100 * time.Millisecond)
+	available := make(chan bool)
+	go func() {
+		for _ = range ticker.C {
```

Comment:
Use construct:
```go
for ;true; <-time.After(time.Duration) {
       ...
       if condition {
           break
       }
}
```
ticker.C will wait for a full tick before trying to do anything (not a huge issue with 100*.time.Millisecond, but if things get retuned to a less aggressive interval, it can become important.

Spawning a go routing and waiting on a channel from that go routine is more complex than the simple for loop above with a break :)

Reply:
Thank you for the suggestion!